addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -2386,7 +2387,8 @@ options.registry.CoverProperties = options.Class.extend({
 
         // TODO: `o_record_has_cover` should be handled using model field, not
         // resize_class to avoid all of this.
-        let coverClass = this.$el.find('[data-cover-opt-name="size"] we-button.active').data('selectClass') || '';
+        let coverClass = this.$el.find('[data-cover-opt-name="size"] we-button.active').data('selectClass') 
+            || this.$el.find('.o_record_cover_opt_size_default').data('selectClass');
```

Comment:
Seems wrong, probably means that this forces the size to change without you seeing it?

Reply:
It does... I kept this class as the default that is applied when there is a change of background. Maybe changing the background should not trigger changing the size, as they are two different options ?

Reply:
Well actually maybe we should not change the size when you change the image indeed. It was required in old version because the "small/fit text" option was not able to have an image but that does not seem to be the case anymore? That won't solve the problem here though, right? Since you simply don't know which size the cover actually uses :grimacing:

Reply:
On the /blog it is possible to display a cover with a background image and a fit text
Maybe a fix on stable could be to overwrite the write method of `website.cover_properties.mixin` to make it smarter and add or not the `o_record_has_cover` class by itself ? So that the dataset attribute `coverClass` would only take care of the `resize_class`, which would be updated only if it has a value
But maybe it is not worth it if we want to refactor it so that it is handled with a new model field in master ?

Reply:
> But maybe it is not worth it if we want to refactor it so that it is handled with a new model field in master ?

Odoo unfortunately don't want to do that (don't ask me why :grimacing:) so we are stuck with a JSON value which forces us to save and override all keys at the same time.

> On the /blog it is possible to display a cover with a background image and a fit text

What do you mean? :thinking: 

> Maybe a fix on stable could be to overwrite the write method of website.cover_properties.mixin to make it smarter and add or not the o_record_has_cover class by itself ?

Can't do that in stable without breaking potential custo

Reply:
Ok, thank you for the explanations. So should I remove this CoverProperties option for the `o_wblog_posts_loop` ?

> > > It was required in old version because the "small/fit text" option was not able to have an image 
> >
> > On the /blog it is possible to display a cover with a background image and a fit text
> 
> What do you mean?

I was answering to that but maybe I didn't fully get the issue that was there in old versions

Reply:
> So should I remove this CoverProperties option for the o_wblog_posts_loop ?

Yes, let's go for that. Change only the JS and `_computeWidgetVisbibility` to hide all cover properties options in case *any* use_XXX variable is false.

Reply:
Should we only do that for the use_size so that it is still possible to edit a cover from the blog post `o_wblog_post_top` ? As the template does not use use_text_align, it would hide all the cover properties options
Also I realize that doing that we won't authorize the user to change the `use_text_align` class anymore as the option is only available on the `opt_blog_cover_post_fullwidth_design`, where `use_size` is false

![image](https://user-images.githubusercontent.com/58561718/120787114-4d8aaa00-c52f-11eb-954f-cd4090ec5dba.png)

Reply:
I don't know the code by heart ^^ Obviously we still want to be able to edit the cover properties from the blog post yes... the point is only to disable edition where it does not work.